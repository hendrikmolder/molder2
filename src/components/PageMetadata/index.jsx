import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'
import { Image } from 'semantic-ui-react'
import Moment from 'react-moment'

const TextContainer = styled.div`
    display: flex;
    flex-direction: column;
    margin-left: 10px;
    line-height: 16px;
    font-size: 14px;
    font-weight: 400;
`

const Container = styled.div`
    display: flex;
    justify-items: center;
`

const StyledMoment = styled(Moment)`
    font-size: 14px;
    font-weight: 400;
    color: rgb(132, 132, 132);
`

const PageMetadata = (props) => {
    const { avatar, name, published_at } = props

    return (
        <Container>
            <Image src={avatar} avatar />
            <TextContainer>
                {name}
                <StyledMoment format="Do MMMM YYYY" date={published_at}/>
            </TextContainer>
        </Container>
    )
}

PageMetadata.propTypes = {
    avatar: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired,
    published_at: PropTypes.string.isRequired
}

export default PageMetadata