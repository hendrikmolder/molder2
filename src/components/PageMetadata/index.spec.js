import React from 'react'
import renderer from 'react-test-renderer'

import PageMetadata from './index'

const NAME = "test"
const DATE = "2019-06-02T14:49:07+0000"
const IMAGE = "https://localhost/image.jpeg"

describe('PageMetadata', () => {
    it('renders correctly', () => {
        const tree = renderer
            .create(<PageMetadata name={NAME} published_at={DATE} avatar={IMAGE} />)
            .toJSON()
        expect(tree).toMatchSnapshot()
    })
})