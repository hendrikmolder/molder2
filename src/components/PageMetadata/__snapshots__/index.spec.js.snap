// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageMetadata renders correctly 1`] = `
<div
  className="sc-bwzfXH ehjSNy"
>
  <img
    className="ui avatar image"
    src="https://localhost/image.jpeg"
  />
  <div
    className="sc-bdVaJa juvHRq"
  >
    test
    <time
      className="sc-htpNat bweAlh"
      dateTime={"2019-06-02T14:49:07.000Z"}
    >
      2nd June 2019
    </time>
  </div>
</div>
`;
